### ISSUE TITLE
Missing warning for bool conversion from pointer

### ISSUE THREAD

**Issue by mkbosmans** [llvm/llvm-project#68401]
When compiling the code
```
#include<stdbool.h>

void getBool(bool out);
void getInt(int out);

void test(bool *b, int* i);
void test(bool *b, int* i) {
  getBool(b);
  getInt(i);
}
```
with `clang -std=c99 -Weverything -c bool_pointer_warning.c`, I get a warning about the pointer argument of getInt, but not about the pointer argument of getBool.

I guess that is because a pointer is implicitly converted to a boolean, but it would be nice to be able to ask to be warned about the forgotten pointer dereference here.

**Comment by shafik** [comment]
I think warning on converting a pointer to bool would be difficult but CC @AaronBallman

**Comment by AaronBallman** [comment]
Well, implementation-wise, it would be easy to do the naive implementation. However, the problem is that implicit conversion from pointer to bool is an *incredibly* common idiom and there's basically no way we could enable the diagnostic by default. We could maybe reduce the chattiness by restricting it only to diagnosing when a pointer to bool is implicitly converted to a bool, but I'm not certain whether that would be sufficient to allow us to enable the diagnostic by default; someone would have to run the check over a large corpus of code to see how many true positives it finds compared to false positives.
